File: 520.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        if(word.size()<=1){
            return true;
        }
        
        if(islower(word[1])){
            for(int i=2;i<word.size();++i){
                if(isupper(word[i])){
                    return false;
                }
            }
            return true;
        }
        
        for(int i=0;i<word.size();++i){
            if(islower(word[i])){
                return false;
            }
        }
        
        return true;
    }
};

File: Python/520.py
class Solution(object):
    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        if word == "":
            return True
        if word.islower():
            return True
        if word.isupper():
            return True
        if word[0].isupper() & word[1:].islower():
            return True
        return False
